warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/DelayedFieldIndexIterator.java

```diff
@@ -0,0 +1,112 @@
+package datawave.query.postprocessing.tf;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.WrappingIterator;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Iterator that fetches all keys defined by the seek range. Used by the {@link TermFrequencyHitFunction} to determine if a field value pair is present in a
+ * document. In the TLD case this iterator will also fetch all child uids associatd with the TLD.
+ */
+public class DelayedFieldIndexIterator extends WrappingIterator {
```

Comment:
As discussed, consider dropping this entirely? It's just a wrapper around a source and doesn't do anything special. If we do indeed just need a wrapper, perhaps there's another existing implementation of `WrappingIterator` we can use so we don't need to implement and maintain this one?

Reply:
Dropping like a hot potato! Shame that the original idea didn't pan out, but on the bright side less code to be maintained.